tests/x11/rstudio_server.pm

```diff
@@ -0,0 +1,67 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+# Summary: Basic test of RStudio Server
+# Maintainer: Dan Čermák <dcermak@suse.com>
+
+use base "x11test";
+use strict;
+use warnings;
+use testapi;
+use utils;
+use rstudio;
+
+sub run() {
+    mouse_hide(1);
+    ensure_installed('rstudio-server');
+
+    x11_start_program('xterm');
+    become_root();
+    assert_script_run("systemctl start rstudio-server");
+    wait_still_screen(1);
+    send_key("alt-f4");
+
+    # open R Studio in the browser and don't verify that Firefox actually
+    # started via a needle
+    x11_start_program("firefox http://localhost:8787", valid => 0);
+    wait_still_screen;
+
+    # login
+    assert_and_click("rstudio_server-login-username");
+    # username
+    type_string('bernhard');
+    # password
+    send_key('tab');
+    type_password;
+    send_key('ret');
+
+    # reject saving the password for bernhard but don't fail in case this doesn't show up
+    check_screen('rstudio_server-firefox-save-password-prompt', timeout => 10) && assert_and_click('rstudio_server-firefox-save-password-prompt', timeout => 1);
```

Comment:
Please use a multi-tag `assert_screen` with `match_has_tag` instead of `check_screen` with non-zero timeout to prevent introducing any timing dependant behaviour, to save test execution time as well as state more explicitly from the testers point of view what are the expected alternatives. For example:

```perl
assert_screen([qw(yast2_console-finished yast2_missing_package)]);
if (match_has_tag('yast2_missing_package')) {
    send_key 'alt-o';  # confirm package installation
    assert_screen 'yast2_console-finished';
}
```